### ISSUE TITLE
Add headings for the Editor screen main sections

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#503]
Splitting this out from https://github.com/WordPress/gutenberg/issues/467

In the current plugin implementation, the Editor screen has no headings at all.

![screen shot 2017-04-25 at 00 01 36](https://cloud.githubusercontent.com/assets/1682452/25361553/1af79902-294f-11e7-9e49-0bf01e992fe8.jpg)

It misses a main H1 and maybe it could benefit from some H2 used to identify the main UI sections (to evaluate, this depends on the actual grouping of the main controls).

Headings are used for content navigation and allow screen reader users to "jump" from a section to another one. That's clearly explained on https://make.wordpress.org/core/2015/10/28/headings-hierarchy-changes-in-the-admin-screens/ which also illustrates all the work done about headings in the last releases.
A proper headings hierarchy is also a WCAG requirement. 

I'm not a designer, so I'd leave the visual part to designers 🙂 For the accessibility part, I'd propose to discuss this issue in the next accessibility meeting on Slack.

**Comment by jasmussen** [comment]
We haven't designed a spot for this heading, so the idea of adding it as invisible screen reader text appeals to me. What tool would you recommend we use when going about this?

**Comment by afercia** [comment]
I'd also note that, currently, the main heading is the only piece of information that says to user _what_ they're editing: post, custom post, or page:

<img width="540" alt="post" src="https://cloud.githubusercontent.com/assets/1682452/25535460/6c6d377a-2c37-11e7-9352-06ee803217d8.png">
<img width="540" alt="project" src="https://cloud.githubusercontent.com/assets/1682452/25535462/6c8e10bc-2c37-11e7-9a15-efc47ea262dc.png">
<img width="490" alt="page" src="https://cloud.githubusercontent.com/assets/1682452/25535461/6c75708e-2c37-11e7-8df9-61ef7dc98672.png">

**Comment by afercia** [comment]
> What tool would you recommend we use when going about this?

To hide text in a way that's still accessible, WordPress uses the `screen-reader-text` CSS class. However, I'd consider to make the heading visible as per the consideration in my last comment.

**Comment by mapk** [comment]
I'm moving my comment from #467 , sorry for derailing the other ticket.

![heading](https://cloud.githubusercontent.com/assets/617986/25537282/b1632406-2bf2-11e7-8e84-3efdb75f597c.png)

I've added the word "Editor:" to the Toolbar. That word could be an H1, but styled to match the surrounding interface. The dropdown "Visual" would not be part of it, and not affect the H1. Seems like it might fit nicely with tab order too.

Any thoughts?

**Comment by jasmussen** [comment]
I really think we should explore the hidden section title here. In my mind, the active section is already highlighted both by the page `<title>`, as well as the selected item in the sidebar. I don't know how you can end up on this page, without knowing what it's all about. In addition to this, the sidebar button label will indicate whether you're editing a post, page, or custom post type —  "Page Settings".

**Comment by afercia** [comment]
I think there's the risk of underestimating the potential confusion for users here 🙂  Having a cleaner UI is a win for everyone. But, as I see it, this shouldn't happen at the cost of removing important information.

I still think there should be some sort of indication about the "what", together with re-considering the ordering and grouping of controls. I'd also like to see this being discussed by the design team and be open to the community feedback. As far as I know, a real, broad, discussion hasn't happened yet.

One more potential issue is about controls that use icons only. Icons don't have a universal meaning. I don't fully understand why some controls have only icons and other have icons plus some visible text. What't the reasoning behind that?

Also to consider, there are functions in WP that provide labels to be used by CPTs for this screen. Integrating such a change wouldn't be so easy.

**Comment by joyously** [comment]
>  I don't know how you can end up on this page, without knowing what it's all about.

Are you thinking of users that only ever use one browser window? There are a lot of users that always use more than one window, especially when copying. In a scenario that involves combining content from multiple pages or custom posts into a different post, there could be several windows open on editor pages that all look similar (due to copied content). How would that user tell them apart easily?

**Comment by afercia** [comment]
In the current WP editor, one of the most difficult tasks for screen reader users is navigating from the editable area to the editor toolbar. There is the TinyMCE shortcut `Alt+F10`, maybe not so intuitive. Instead, headings are the [predominant mechanism for finding content](http://webaim.org/projects/screenreadersurvey5/#finding), together with an increasing knowledge and use of ARIA landmark regions.

I've played a bit with this, and a few headings could greatly help, for example using Safari + VoiceOver:

![headings test](https://cloud.githubusercontent.com/assets/1682452/25555836/dee2847c-2cf0-11e7-84cb-f863ab790830.jpg)

It would be a matter of just a few keypresses to find the toolbars or skip the toolbars and go straight to the editor (headings text is just for testing purposes, worth considering a better wording).

**Comment by afercia** [comment]
To recap the headings situation as of August 3rd:

![screen shot 2017-08-03 at 11 13 39](https://user-images.githubusercontent.com/1682452/28914977-db4b5bde-783d-11e7-910f-a21b6b823f20.png)

The headings in the first group in the screenshot above are all in the post content. The post title is always presented as H1.

The headings in the second group are in the editor sidebar and they start with a H3 level.

Right now, the Gutenberg screen is the only screen in WordPress that doesn't have a main H1 heading to identify the purpose of the page. In the classic editor, this is done with the h1 "Edit Post", "Edit Page", "Edit {custom post type name}".

**Comment by afercia** [comment]
Marking with the high priority label to indicate it's an a11y priority. The Gutenberg page is the only one in WordPress that doesn't have a proper heading hierarchy starting with an H1 that indicates what the page is about.

**Comment by afercia** [comment]
Slightly unrelated but also the document title should be fixed:

old editor:
`<title>Edit Post &lsaquo; WordPress Develop &#8212; WordPress</title>`

(or "Edit Page" etc, depending on what's being edited)

Gutenberg:
`<title>Welcome to the Gutenberg Editor | Add New Post ‹ WordPress Develop — WordPress</title>`
